src/core/election.rs

```diff
@@ -0,0 +1,103 @@
+///     * Voting with an elector not registered during Election's initialization, thrown error: "Elector doesn't exist!".
+///    
+///     # Examples
+///   
+///     ```
+///     use std::hash::Hash;
+///     use crate::core::election::Election;
+///
+///     #[derive(Hash, Eq, Clone, Debug)]
+///     pub enum Candidates {
+///         Candidate_1,
+///         Candidate_2,
+///         Candidate_3,
+///         Candidate_4,
+///     }
+///     let electors = [String::from("elector_1"),
+///                     String::from("elector_2"),
+///                     String::from("elector_3")];
+///     let votes = [Candidates::Candidate_1,
+///                  Candidates::Candidate_2,
+///                  Candidates::Candidate_3];
+///
+///     let mut election = Election::<Candidates>::new(&electors);
+///     for (index, voter) in electors.iter().enumerate() {
+///        election.vote(&voter, votes[index]);
+///     }
+///     let election_result = election.count_votes().ok().unwrap();
+///     println!("{:?}", election_result.result());
+///     ```
+pub struct Election<T> {
+    pub electors_votes: HashMap<String, Option<T>>,
+}
+
+impl<T> Election<T> {
+    pub fn new(electors: &[String]) -> Election<T> {
+        if electors.len() == 0 {
+            panic!("No electors in this election!");
+        }
+        let electors_votes: HashMap<String, Option<T>> = (*electors).iter()
+                                                          .map(|v| (v.clone(), None))
+                                                          .collect();
+        Election { electors_votes, }
+    }
+
+    pub fn vote(&mut self, elector: &String, vote: T) {
+        if let Some(elector) = self.electors_votes.get_mut(elector) {
+            elector.get_or_insert(vote);
+        } else {
+            panic!("Elector doesn't exist!");
+        }
+    }
+
+    pub fn count_votes(self) -> Result<Scrutiny<T>, Election<T>> 
+    where T: 
+        Eq + Hash + Clone,
+    {
+        let elector_not_having_voted = self.electors_votes.values()
+                                                          .filter(|v| !v.is_some())
+                                                          .count();
+        if elector_not_having_voted > 0 {
+            Err(self)
+        } else {
+            let mut votes_counter = HashMap::<Option<T>, usize>::new();
+            for vote in self.electors_votes.values() {
+                if let Some(val) = votes_counter.get_mut(&vote) {
+                    *val += 1;
+                } else {
+                    votes_counter.insert(vote.clone(), 1);
+                }
+            }
+            Ok(Scrutiny { result: votes_counter, })
+        }
+    }
+}
+
+pub struct Scrutiny<T> {
+    result: HashMap::<Option<T>, usize>,
+}
+
+impl<T> Scrutiny<T> {
+    pub fn result(&self) -> &HashMap::<Option<T>, usize> {
+        &self.result
+    }
+}
```

Comment:
These files shouldn't be in the PR, maybe rebase?

Reply:
indeed, i'll rebase